src/native/managed/cdacreader/src/Contracts/Metadata.cs

```diff
@@ -0,0 +1,386 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection.Metadata.Ecma335;
+using UntrustedMethodTable = Microsoft.Diagnostics.DataContractReader.Contracts.UntrustedMethodTable_1;
+using MethodTable = Microsoft.Diagnostics.DataContractReader.Contracts.MethodTable_1;
+using UntrustedEEClass = Microsoft.Diagnostics.DataContractReader.Contracts.UntrustedEEClass_1;
+using EEClass = Microsoft.Diagnostics.DataContractReader.Contracts.EEClass_1;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+internal interface IMetadata : IContract
+{
+    static string IContract.Name => nameof(Metadata);
+    static IContract IContract.Create(Target target, int version)
+    {
+        TargetPointer targetPointer = target.ReadGlobalPointer(Constants.Globals.FreeObjectMethodTable);
+        TargetPointer freeObjectMethodTable = target.ReadPointer(targetPointer);
+        return version switch
+        {
+            1 => new Metadata_1(target, freeObjectMethodTable),
+            _ => default(Metadata),
+        };
+    }
+
+    public virtual MethodTable GetMethodTableData(TargetPointer targetPointer) => throw new NotImplementedException();
+
+    public virtual TargetPointer GetClass(in MethodTable methodTable) => throw new NotImplementedException();
+
+    public virtual bool IsString(in MethodTable methodTable) => throw new NotImplementedException();
+    public virtual bool ContainsPointers(in MethodTable methodTable) => throw new NotImplementedException();
+    public virtual bool IsDynamicStatics(in MethodTable methodTable) => throw new NotImplementedException();
```

Comment:
I don't understand yet what this flag means or why SOS cares about it

Reply:
Is it so that all statics are not considered dynamic (since https://github.com/dotnet/runtime/pull/99183 `All statics are now considered "dynamic" statics.`). If it's representing the same data, then it can be hardwired return `true`?

Reply:
yea things will be simpler in this respect in 99183.  I need to update to sync things up